x-pack/legacy/plugins/siem/public/pages/detection_engine/create_rule/components/mitre/index.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiFormRow,
+  EuiSelect,
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiComboBox,
+  EuiFormControlLayout,
+} from '@elastic/eui';
+import { isEmpty, kebabCase, camelCase } from 'lodash/fp';
+import React, { ChangeEvent, useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+
+import { tacticsOptions, techniquesOptions } from '../../../mitre/mitre_tactics_techniques';
+import * as CreateRuleI18n from '../../translations';
+import { FieldHook, getFieldValidityAndErrorMessage } from '../shared_imports';
+import * as I18n from './translations';
+import { IMitreEnterpriseAttack } from '../../types';
+
+const MyEuiFormControlLayout = styled(EuiFormControlLayout)`
+  .euiComboBox__inputWrap {
+    height: 32px !important;
+    overflow-y: scroll;
+  }
+`;
+interface AddItemProps {
+  field: FieldHook;
+  dataTestSubj: string;
+  idAria: string;
+  isDisabled: boolean;
+}
+
+export const AddMitreThreat = ({ dataTestSubj, field, idAria, isDisabled }: AddItemProps) => {
+  const { isInvalid, errorMessage } = getFieldValidityAndErrorMessage(field);
+
+  const removeItem = useCallback(
+    (index: number) => {
+      const values = field.value as string[];
+      field.setValue([...values.slice(0, index), ...values.slice(index + 1)]);
+    },
+    [field]
+  );
+
+  const addItem = useCallback(() => {
+    const values = field.value as IMitreEnterpriseAttack[];
+    const { id, name, reference } = tacticsOptions[0];
+    if (!isEmpty(values[values.length - 1])) {
+      field.setValue([...values, { tactic: { id, name, reference }, techniques: [] }]);
+    } else {
+      field.setValue([{ tactic: { id, name, reference }, techniques: [] }]);
+    }
+  }, [field]);
+
+  const updateTactic = useCallback(
```

Comment:
Will probably want to clear the selected `Techniques` when the Tactic selection changes, as currently you can select Techniques, change the Tactic, and thus have invalid Techniques and the form says it's valid.